### PR TITLE
Generate a code coverage report

### PR DESCRIPTION
**Pull request by ferrarimarco**
## Proposed Changes

1. Use [kcov](https://github.com/SimonKagstrom/kcov) to generate a code coverage report.
2. Add a Makefile so that we can easily run workflow actions without having to copy/paste from the workflow YAML. Starting with a brand new `test` target.
3. Add a placeholder for the script we're going to use to launch the test suite
4. [Install the Codacy app](https://docs.codacy.com/getting-started/getting-started-with-codacy/)
5. Upload the code coverage data to a code coverage service using the [Codacy GitHub action](https://docs.codacy.com/coverage-reporter/alternative-ways-of-running-coverage-reporter/#github-action)
6. Add [Codacy badges](https://docs.codacy.com/repositories/badges/) to the README.
7. Add a missing "change directory" parameter to a `git` invocation that we use to get the list of changed files.
8. Fix file and directory ownership after testing the action locally.

### Author/Contributor
- [x] If documentation is needed for this change, has that been included in this pull request

### Reviewing Maintainer
- [x] Label as `breaking` if this is a large fundamental change
- [x] Label as either `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`, or `performance`

### CODE REVIEW

**Comment by ferrarimarco** [comment]
@zkoppert any news about the Codacy app? Thanks!

**Comment by zkoppert** [comment]
I prompted the security review team, its still sitting in pending mode

**Review by pap0135 — APPROVED** [review]

**Comment by zkoppert** [comment]
Hey @ferrarimarco, what was next on this one? Need anything from me?

**Comment by admiralAwkbar** [comment]
@ferrarimarco Let me know if I can do anything to help on this!

**Comment by ferrarimarco** [comment]
Hi! Finally had a few minutes to check the codacy side. Let me see if I can finally upload a coverage report now that we set up the codacy side.

**Comment by admiralAwkbar** [comment]
@ferrarimarco notice the inline linters were unhappy with some line lengths.. so I got that worked up :)

**Comment by ferrarimarco** [comment]
Great, thanks!

**Comment by admiralAwkbar** [comment]
@ferrarimarco I added the badge to the readme, and then tried to get the `codecoverage` report to work but it seems we still have some more work to do...
let me know how I can help :)

File: .github/workflows/deploy-DEV.yml [hunk 1]
```diff
@@ -99,12 +101,43 @@ jobs:
...
+      ##################
+      # Run test cases #
+      ##################
+      - name: Run the test suite
+        shell: bash
+        run: |
+          pwd
+          whoami
+          ls -la
+          sudo mkdir -p test/reports/cobertura
```
Comment (ferrarimarco) [inline .github/workflows/deploy-DEV.yml:113]: This should be in the make target :)

Comment (ferrarimarco) [reply .github/workflows/deploy-DEV.yml:113]: LOL don't worry. I'll have a look

**Comment by ferrarimarco** [comment]
Ok, I spotted the issue. Fixing now...

**Comment by ferrarimarco** [comment]
I think the` Test local action` step messes up the permissions. Will have a deeper look.

**Pull request merged**